dracut/30afterburn/afterburn-hostname.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Afterburn Hostname
+# Azure needs to fetch the hostname in the initramfs
+ConditionKernelCommandLine=|ignition.platform.id=azure
+ConditionFirstBoot=yes
+After=ignition-disks.service
+Before=ignition-files.service
```

Comment:
Minor: maybe a comment there that we're purposely ordering before `ignition-files.service` so that the latter can still override the hostname if wanted?

Reply:
Added some comments